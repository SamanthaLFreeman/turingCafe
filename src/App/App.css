@import url("https://fonts.googleapis.com/css?family=Pacifico");

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-image: url("../assets/cafe-img.jpeg");
  width: 100%;
  height: 100vh;
  background-repeat: no-repeat;
  background-size: cover;
  text-align: center;
}

.app-title {
  color: teal;
  font-family: "Pacifico", cursive;
  font-size: 50px;
}

.sort-btn {
  background-color: teal;
  border: solid 1px #fff;
  color: #fff;
  font-size: 1em;
  margin-bottom: 15px;
  padding: 6px;
  width: 14vw;
}
