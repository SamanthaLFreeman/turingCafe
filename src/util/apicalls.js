export const getReservations = () => {
  return fetch('http://localhost:3001/api/v1/reservations')
    .then(response => {
      if (!response.ok) {
        throw new Error('Could not retrieve reservations.')
      }
      return response.json()
    })
}

export const postReservation = (newReservation) => {
  let optionsObj = {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(newReservation),
  }
  return fetch('http://localhost:3001/api/v1/reservations', optionsObj)
    .then(response => {
      if (!response.ok) {
        throw new Error('Could not post a new reservation.')
      }
      return response.json()
    })
}

export const deleteReservation = (id) => {
  const optionsObj = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  return fetch(`http://localhost:3001/api/v1/reservations/${id}`, optionsObj)
    .then(response => {
      if (!response.ok) {
        throw new Error('Could not delete a reservation.')
      }
      return response.json()
    })
}