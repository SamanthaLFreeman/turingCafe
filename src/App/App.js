import React, { Component } from 'react';
import './App.css';
import Reservations from '../Reservations/Reservations'
import ReserveForm from '../ReserveForm/ReserveForm';

class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: []
    }
  }

  componentDidMount() {
    fetch('http://localhost:3001/api/v1/reservations')
      .then(resp => resp.json())
      .then(data => this.setState({reservations: data}))
      .catch(error => console.log(error))
  }

  addReservation = (newReservation) => {
    this.setState({reservations: [...this.state.reservations, newReservation]})
  }

  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
          <ReserveForm addReservation={this.addReservation} />
        </div>
        <div className='resy-container'>
          <Reservations reservations={this.state.reservations} />
        </div>
      </div>
    )
  }
}

export default App;
