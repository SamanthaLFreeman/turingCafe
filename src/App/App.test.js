import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { getReservations, postReservation, deleteReservation } from '../util/apicalls';

jest.mock('../util/apicalls');

getReservations.mockImplementation(() => {
  return Promise.resolve([{
    name: 'Sam',
    date: '1/14',
    time: '16:00',
    number: 2,
    id: 1
  }])
});

postReservation.mockImplementation(() => {
  return Promise.resolve({
    name: 'Sam',
    date: '1/14',
    time: '16:00',
    number: 2,
    id: 1
  })
});

deleteReservation.mockImplementation( () => {
  return Promise.resolve([
    {
      name: 'Sam',
      date: '1/14',
      time: '16:00',
      number: 2,
      id: 1
    }
  ])
})

describe('App', () => {
  let wrapper;

  beforeEach( () => {
    wrapper = shallow(<App />)
  })

  it('should match the snapshot with all of the data passed through', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call postReservation when addReservation is given a new reservation', async () => {
    const reservationMock = {
      name: 'Sam',
      date: '1/14',
      time: '16:00',
      number: 2,
      id: 1
    }
    wrapper.instance().addReservation(reservationMock);
    expect(postReservation).toHaveBeenCalledWith(reservationMock);
  });

  it('should call getReservations when mounting', () => {
    expect(getReservations).toHaveBeenCalled();
  });

  it('should call deleteReservation when cancelReservation is given an id to remove', async () => {
    const reservationId = 2
    wrapper.instance().cancelReservation(reservationId);
    expect(deleteReservation).toHaveBeenCalledWith(reservationId);
  });

  
})
