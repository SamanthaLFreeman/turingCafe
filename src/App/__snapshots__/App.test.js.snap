// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all of the data passed through 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <ReserveForm
      addReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <Reservations
      reservations={
        Array [
          Object {
            "date": "1/14",
            "id": 1,
            "name": "Sam",
            "number": 2,
            "time": "16:00",
          },
        ]
      }
    />
  </div>
</div>
`;
